### PR TITLE
Feature/stuff-i-noticed-while-using

### PR DESCRIPTION
**Pull request by janek26**
Fixes a bunch of bugs that I noticed while using
Adds some very tiny features aswell, which imo provide a lot of value

I tried to have clean commits

### CODE REVIEW

File: packages/extension/src/shared/transactions.ts [hunk 1]
```diff
@@ -18,3 +25,16 @@ export interface Transaction extends TransactionRequest {
...
 }
+
+export const compareTransactions = (
+  a: TransactionBase,
+  b: TransactionBase,
+): boolean => a.hash === b.hash && a.account.networkId === a.account.networkId
+
+export function entryPointToHumanReadable(entryPoint: string): string {
+  try {
+    return capitalize(entryPoint.replaceAll("_", " "))
```
Comment (simonheys) [inline packages/extension/src/shared/transactions.ts:36]: As an alternative, I started using [lowerCase](https://lodash.com/docs/4.17.15#lowerCase) yesterday, works with entrypoints that are in camelCase too:

```
import { lowerCase, upperFirst } from "lodash-es"

...
return upperFirst(lowerCase(entryPoint))
...
```

**Review by simonheys — APPROVED** [review]

File: packages/extension/src/shared/transactions.ts (hunk 1 above)
Comment (janek26) [reply packages/extension/src/shared/transactions.ts:36]: that's very cool

File: packages/extension/src/ui/features/accountTokens/tokens.state.ts [hunk 2]
```diff
@@ -164,18 +159,18 @@ export const useTokensWithBalance = (
         subscription.unsubscribe()
       }
     }
-  }, [])
+  }, [tokenAddresses.join(":")])
```
[the remark this answers is not in the document]
Comment (janek26) [reply packages/extension/src/ui/features/accountTokens/tokens.state.ts:162]: I had infinity loops with it 😅

[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/src/ui/features/accountTokens/tokens.state.ts:162]: Wow that is both unexpected and expected at the same time

**Pull request merged**